### DISCUSSION TITLE
Pimcore 10.6 ckeditor Paste-Issue

### DISCUSSION THREAD

**Discussion by phoebusryan** [pimcore/pimcore#17647] [Q&A / Support] [answered]
Hello,
We recently updated our Pimcore from version 5 to 10.6. Since then we have a problem with the Wysiwyg (ckeditor 4.22.1). When we paste a formatted text from Word into a field, list entries (LI) are formatted completely wrong. The editor does not interpret this as a list. This used to work without any problems. I downloaded version 4.22.1 from the official ckeditor website (https://ckeditor.com/cke4/release/CKEditor-4.22.0-4.22.1) and it works there too. What could be the reason for this? We have never modified anything in the editor... except set the version check to false. But that doesn't change anything.

Please help :)

**Comment by kingjia90** [comment]
Just quickly checked 10.6 and in there i don't see any usages of [pastefromword](https://ckeditor.com/cke4/addon/pastefromword) plugin, maybe it's caused by that, it's not shipped with this plugin enabled out of the box and need to be added on your own

**Reply by kingjia90** [reply]
digged a little more and it was removed here
https://github.com/pimcore/pimcore/commit/ad1a9b013a713835f75285dc2b1c522e1df50ec3#diff-f98f26a96ed8a6effc96ea896cc28c4f91b51978d7f844c301971fd195e7c6f3L85-L86
due to security issues found on https://github.com/pimcore/pimcore/issues/3831 but you should be able to re-add it (at own risk)

**Reply by phoebusryan** [reply]
Interesting! So how can I readd it again? (Without impairing the update capability).

Our PIM is only internal and not public so this "should" be okay.

**Reply by kingjia90** [reply]
Haven't tried but i believe with this approach https://ckeditor.com/docs/ckeditor4/latest/guide/dev_plugins.html#manual-installation should work

**Reply by phoebusryan** [reply, marked as answer]
Hm, I tried that and it does not work. The pluginfolder 'pastefromword' is missing. I think I have to create a custom build but don't know how at the moment.

[Edit]
Okay found a solution:
- download https://ckeditor.com/cke4/addon/pastetools 
- download https://ckeditor.com/cke4/addon/pastefromword
- unzip them in /pimcore/bundles/AdminBundle/Resources/public/js/lib/ckeditor/plugins
- open /pimprod/public/bundles/pimcoreadmin/js/lib/ckeditor/config.js
- add config.extraPlugins = 'pastefromword';

this works perfectly
